platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,1001 @@
+ */
+message TssMessageTransaction {
+
+  /**
+   * The node id is the id of the node that generated the TssMessage.
+   */
+  uint64 node_id = 1;
+
+  /**
+   * The source roster hash is the hash of the roster containing the node generating the TssMessage.
+   */
+  bytes source_roster_hash = 2;
+
+  /**
+   * The target roster hash is the hash of the roster that the TssMessage is for.
+   */
+  bytes target_roster_hash = 3;
+
+  /**
+   * The share index establishes a global ordering of shares across all shares in the network.  It corresponds to the
+   * index of the public share in the list returned from the TSS library when the share was created for the source
+   * roster.
+   */
+  uint64 share_index = 4;
+
+  /**
+   * The tss_message is the raw bytes of the TssMessage generated by the node for the share_index.
+   */
+  bytes tss_message = 5;
+
+  /**
+   * The signature is produced by the node signing the tuple (source_roster_hash, target_roster_hash, share_index,
+   * tss_message) using its private `tssEncryptionKey`.  The signature is verified by other nodes using the public
+   * key of the `tssEncryptionKey` of the node sending this transaction.
+   */
+  bytes signature = 6;
+}
+```
+
+##### TssVote System Transaction
+
+```protobuf
+/**
+ * The TssVoteTransaction is used to vote on the validity of TssMessages for a candidate roster.
+ */
+message TssVoteTransaction {
+
+  /**
+   * The node id is the id of the node that generated the TssVote.
+   */
+  uint64 node_id = 1;
+
+  /**
+   * The source roster hash is the hash of the roster containing the node generating the TssVote.
+   */
+  bytes source_roster_hash = 2;
+
+  /**
+   * The target roster hash is the hash of the roster that the TssVote is for.
+   */
+  bytes target_roster_hash = 3;
+
+  /**
+   * The ledger id that was computed from the TssMessages for the target roster.
+   */
+  bytes ledger_id = 4;
+
+  /**
+   * The signature produced by the node using its RSA signing key to sign the ledger_id.  This signature is used to
+   * establish a chain of trust in the ledger id.
+   */
+  bytes node_signature = 5;
+
+  /**
+   * The tss_vote is a bit vector where the index position of each bit corresponds to the sequence number of a
+   * `TssMessageTransaction`. The least significant bit of byte[0] corresponds to sequence number 0 and the most
+   * significant bit of byte[0] corresponds to sequence number 7.  The least significant bit of byte[1] corresponds to
+   * sequence number 8 and the most significant bit of byte[1] corresponds to the sequence number 15, and so on.
+   * If a bit is set to 1, then the TssMessage for the corresponding TssMessageTransaction sequence number was
+   * received as one of the first Threshold number of valid TssMessages in consensus order.  If a bit is set to 0,
+   * then the TssMessage was invalid or there was no TssMessageTransaction with a sequence number corresponding to
+   * the bit index.
```

Comment:
Two items here:
1. The indentation should be 4 spaces, not 2 for all protobufs.
1. This particular description is exceptionally detailed; it _may_ be a bit
   more than is needed.
```suggestion
    /**
     * A bit vector where the index position of each bit corresponds to the
     * sequence number of a `TssMessageTransaction`. The least significant bit
     * of byte[0] corresponds to sequence number 0 and the most significant bit
     * of byte[0] corresponds to sequence number 7.  The least significant bit
     * of byte[1] corresponds to sequence number 8 and the most significant bit
     * of byte[1] corresponds to the sequence number 15, and so on. If a bit is
     * set to 1, then the TssMessage for the corresponding TssMessageTransaction
     * sequence number was received as one of the first Threshold number of
     * valid TssMessages in consensus order.  If a bit is set to 0, then the
     * TssMessage was invalid or there was no TssMessageTransaction with a
     * sequence number corresponding to the bit index.
     */
```